### ISSUE TITLE
Post-hoc remove of external joiners

### ISSUE THREAD

**Issue by bifurcation** [mlswg/mls-architecture#152]
_Comment by @paulwouters_

~~~ 
   External joins do
   not allow for more granular authorization checks to be done before
   the new member is added to the group, so if an application wishes to
   both allow external joins and enforce such checks, then the
   application will need to do such checks when a member joins, and
   remove them if checks fail.
~~~
This seems rather risky? What could such a member do in the short time between getting added and getting kicked? Would they see message history? Group membership? How many devices a user has? When members were last online? Why support such a group type? Wouldn't it be better to not support this and have an automated "special member" being a bot that handles member addition?

**Comment by bifurcation** [comment]
Until someone else sends a message, a new joiner doesn't learn anything non-public.  If you think about it, how would they?  They're outside the group before they send an external join, and the external join just establishes a new shared secret with the group for future communications.

More specifically:
* Message history - MLS's forward secrecy guarantees mean that new joiners don't get message history.
* Group membership - This is public anyway via the GroupInfo that an external joiner needs to join
* How many devices - To the degree this is exposed, it would be in the group membership
* Last online - MLS doesn't have any notion of this

As for an "add me bot", it's a useful idea, but not applicable to a bunch of cases.  The bot has to be a member of the group and has to be always-online.  So now you have the group's secrets on a server somewhere, instead of just in the communicating endpoints, already an additional point of compromise.  And it can't be the DS's server, since the whole point is to protect against the DS.

**Comment by rohan-wire** [comment]
> _Comment by @paulwouters_
> 
> ```
>    External joins do
>    not allow for more granular authorization checks to be done before
>    the new member is added to the group, so if an application wishes to
>    both allow external joins and enforce such checks, then the
>    application will need to do such checks when a member joins, and
>    remove them if checks fail.
> ```

This statement in the draft is just plain wrong. In our implementation, we actually do validation of External joins at the application layer. I addressed this in [PR#116](https://github.com/mlswg/mls-architecture/pull/116) (dating from WGLC) which says the following:

"External joins do not allow for more granular authorization checks to be done before the new member is added to the group, so if an application wishes to both allow external joins and enforce such checks, then either all
the members of the group must all have the ability to check and reject invalid External joins autonomously, or the application needs to do such checks when a member joins and remove them if those checks fail."
